### ISSUE TITLE
Change styling for chat message links

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#1551]
Right now links are a dimmer version of regular text. It's a bit confusing, because if anything you'd expect links to be stronger and/or more opaque.

![image](https://user-images.githubusercontent.com/414923/143184934-c517fa62-6be6-4b60-850e-1704e9c3221a.png)

**Comment by UXShawrk** [comment]
The design is pretty confusing. As a user, I'm not actually sure which ones are links. I assume the links are SONG LIST, Stripe, PayPal, and Bitcoin? I also assume "Donate with" is a drop down?

Where can I find this in the website, by the way?

**Comment by gabek** [comment]
The above screenshot is somebody's server, but you can just send a link in a chat message to see what a link looks like. If you want to link some text like in the screenshot you can use markdown like 
```[my link](http://duckduckgo.com)```

**Comment by UXShawrk** [comment]
Hi, I'm proposing some potential redesign to the texts in order to show links better. Pay no attention to the other parts of the image like the icons or the font, this is to show a little tweak that can make the user know when something is a link.

Conventionally, users read underlined text as a link so that's what I did here. I have three redesign options:

Option 1:
![image](https://user-images.githubusercontent.com/96269733/147953044-df7b8f9e-7bee-4cf5-94b3-f87c86ae6e3f.png)
The links are underlined and coloured blue, which is the conventional colour for links. Darker blue is for links that have already been clicked.

Option 2:
![image](https://user-images.githubusercontent.com/96269733/147955800-4fe105fe-a297-496a-8e7a-444dad24801d.png)
Since the background is a shade of blue, maybe option 1's designs wouldn't be so visible. Here the text is a uniform white but the links are underlined. A more greyish colour is for a link that's been clicked.

Option 3:
![image](https://user-images.githubusercontent.com/96269733/147956111-dbcdcfaf-2a5b-4672-b8e1-421a5ff84880.png)
Same as option 2, but you put in arrow icons on the right to indicate to the user that the text is a link.


---
Tell me what you think and I'll give the hex codes for the colours if you need them.

**Comment by gabek** [comment]
Thanks for looking at this! I have some thoughts.

Since chat message background colors are varied depending on the user sending them, I think the color will always have to be some version of white so it's readable. Otherwise you might end up with the blue on top of blue that you're starting to see in your first example.

Admittedly I should have put multiple examples of chat messages in this issue, but I just assumed people were used to knowing what Owncast chat messages looked like.

![image](https://user-images.githubusercontent.com/414923/147959214-588c9483-8f4e-43c2-b132-ab48745434ac.png)

For most messages links will be inline, not each in its own line. That being said, I do like the icon idea, but I think using a chevron might make people would wonder why there's a "greater than" in the message. But something that's specific to links might be the answer. Like whatever that little box with the arrow is that signifies you're linking somewhere? I'm not sure what that's called.

Would the post-link icon be enough to signify something as a link with an on-hover underline? Without requiring always being underlined?

I'm not the biggest fan of having a "clicked" color, since in this case the "Bitcoin" clicked color just kind of looks like the color of links as they exist today, a dulled out color that just kind of looks broken. Especially with these system messages you'll see them every time you join the chat, so you'd see them dulled out more than you'd see them at their correct color.

**Comment by UXShawrk** [comment]
An on-hover underline wouldn't be a bad idea. In any case, if you do decide to implement underlines being visible only when a mouse hovers over the link, it's good practice to design the link's visuals so that it can be easily differentiated from the rest of the text.

That's why an underlined blue font works so well for links across the web. Most of the time you see black text, so when a link is blue and underlined it really stands out and tells the user "hey, I'm different from the rest of the text!"

Basically, if you want the user to easily identify a link, you need to make it stand out a bit from the rest of the text. Having a dimmer colour may be alright for indicating a link since the "https" is already a signal to most users (plus slightly different colour) that it's a link; but personally, I think you can design it so that it can stand-out more.

**Comment by gabek** [comment]
I experimented with putting the little link icon next to it, but that looks bad.

![image](https://user-images.githubusercontent.com/414923/148330828-797e5a89-09cc-4f19-a752-6281f6ba49a0.png)

The reason I didn't want blue text is for the cases when the text is blue on blue, like this, it's really hard to read.

![image](https://user-images.githubusercontent.com/414923/148330947-28f5cb27-961c-4c4f-b0de-3c67699b2b5c.png)

I caved and went for a translucent underline, so that's probably the best bet. It's readable and it's obvious it's a link.  So good enough?

![image](https://user-images.githubusercontent.com/414923/148331292-005b83a3-9f40-4968-ac00-f9fcbf43a837.png)
